Source/JavaScriptCore/dfg/DFGAbstractInterpreterInlines.h

```diff
@@ -3134,14 +3134,15 @@ bool AbstractInterpreter<AbstractStateType>::executeEffects(unsigned clobberLimi
                 structure = globalObject->nullPrototypeObjectStructure();
             else if (base.isObject()) {
                 // Having a bad time clears the structureCache, and so it should invalidate this structure.
-                bool isHavingABadTime = globalObject->isHavingABadTime();
-                // Normally, we would always install a watchpoint. In this case, however, if we haveABadTime, we
+                bool isStructureCacheCleared = globalObject->isStructureCacheCleared();
```

Comment:
nit: This is a weird name. Why not just check the watchpoint if it's still valid? Since if this returns true, it doesn't mean the cache is cleared right now, it just means we were cleared at some point. I think the code is easier to reason about if we just ask if the watchpoint is still valid.

Reply:
Sounds good. I'll add a helper.